addons/google_gmail/models/google_gmail_mixin.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import requests
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+from odoo.addons.google_account.models.google_service import GOOGLE_TOKEN_ENDPOINT, TIMEOUT
+
+
+class GoogleGmailMixin(models.AbstractModel):
+
+    _name = 'google.gmail.mixin'
```

Comment:
Why would we have to update google drive ? We can inherit from google service with another name, meaning we don't add stuff on google service itself. That is why I had in mind.

Reply:
But if `google.gmail.mixin` inherits from `google.service` we will need to change `generate_refresh_token` e.g. to take the service name from `self.service` instead of taking it from the method argument, which means that the call for gdrive will fail.

Or we keep the service name in the argument (and not as a model attribute), but here the only advantage is to use `self` instead of `self.env['google.service']`

Reply:
Why would it be complicated to update google drive ? Not sure to understand.

Reply:
because there's no model for google drive

Here we give the service name "drive" in the method attribute `addons/google_drive/models/res_config_settings.py:32`, but you propose to inherit from `google.service` and to put the service name in a class variable (`_google_service_name='gmail'`), so the code for drive will not work anymore...

It's more flexible to keep the service as a method attribute (otherwise you will need to create a class for each service, just to define the `_google_service_name`)

Reply:
Arf ... actually this should not be an odoo model then but only tools. Ok then, drop that.